end-to-end-tests/src/bin/bootstrap.rs

```diff
@@ -0,0 +1,100 @@
+use anyhow::{bail, Result};
+use end_to_end_tests::helpers::ctx::{build_client, nexus_addr, Context};
+use end_to_end_tests::helpers::generate_name;
+use omicron_test_utils::dev::poll::{wait_for_condition, CondCheckError};
+use oxide_client::types::{
+    ByteCount, DiskCreate, DiskSource, IpPoolCreate, IpRange, Ipv4Range,
+};
+use oxide_client::{ClientDisksExt, ClientOrganizationsExt, ClientSystemExt};
+use std::net::IpAddr;
+use std::time::Duration;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let client = build_client()?;
+
+    // ===== ENSURE NEXUS IS UP ===== //
+    eprintln!("waiting for nexus to come up...");
+    wait_for_condition(
+        || async {
+            client
+                .organization_list()
+                .send()
+                .await
+                .map_err(|_| CondCheckError::<oxide_client::Error>::NotYet)
+        },
+        &Duration::from_secs(1),
+        &Duration::from_secs(300),
+    )
+    .await?;
+
+    // ===== CREATE IP POOL ===== //
+    eprintln!("creating IP pool...");
+    let nexus_addr = match nexus_addr().ip() {
+        IpAddr::V4(addr) => addr.octets(),
+        IpAddr::V6(_) => bail!("not sure what to do about IPv6 here"),
+    };
+    // TODO: not really sure about a good heuristic for selecting an IP address
+    // range here. in both my (iliana's) environment and the lab, the last octet
+    // is 20; in my environment the DHCP range is 100-249, and in the buildomat
+    // lab environment the network is currently private.
+    let first = [nexus_addr[0], nexus_addr[1], nexus_addr[2], 50].into();
+    let last = [nexus_addr[0], nexus_addr[1], nexus_addr[2], 90].into();
```

Comment:
Nit, can we at least move this "IP range fro the pool" construction to the "helpers"?

This seems like it'll definitely need to be changed as we continue updating how nexus gets launched, but it should probably be coupled tightly with "what network setup we used for nexus.

Reply:
Move it to the end-to-end helpers so that other test code could potentially get the first/last address in the pool that was set up? Or another use case (or somewhere else entirely)?

I can do this but I'm curious about why it would be necessary -- a test could also list the system IP pools if needed, which would probably be more reliable than parsing a config file that can potentially change / go away.

Reply:
My recommendation here was really small - basically just to move this to a function called `get_system_ip_pool` or something, so that the hack is isolated and can easily be replaced. I had no opinion on re-using this function within test bodies themselves -- that didn't seem necessary yet.